components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -2976,9 +2976,17 @@ public Set<AccessTokenDO> getAccessTokensByBindingRef(String bindingRef) throws
                             String subjectIdentifier = resultSet.getString("SUBJECT_IDENTIFIER");
                             String userDomain = resultSet.getString("USER_DOMAIN");
                             String authenticatedIDPName = resultSet.getString("NAME");
+                            String authorizedOrganization = resultSet.getString("AUTHORIZED_ORGANIZATION");
                             AuthenticatedUser user = OAuth2Util.createAuthenticatedUser(authzUser,
                                     userDomain, OAuth2Util.getTenantDomain(tenantId), authenticatedIDPName);
                             user.setAuthenticatedSubjectIdentifier(subjectIdentifier);
+                            if (user.isFederatedUser() &&
+                                    !OAuthConstants.AuthorizedOrganization.NONE.equals(authorizedOrganization)) {
+                                user.setAccessingOrganization(authorizedOrganization);
+                                user.setUserResidentOrganization(resolveOrganizationId(user.getTenantDomain()));
+                                // Tenant domain should be properly set for the organization SSO login users.
```

Comment:
add a comment explaining why change the tenant.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2251/commits/68f107b71a409dc695fd30b6f5c6b462086fae0c